### PR TITLE
Set priorities to 'creation-requested' and 'creation-in-progress'

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

We are setting priorities on the following states while executing

- creation-requested
- creation-in-progress

#### Ticket Link

  Fixes [MM-33114](https://mattermost.atlassian.net/browse/MM-33114)

#### Release Note
```release-note
Set priorities to 'creation-requested' and 'creation-in-progress'
```

### CODE REVIEW

File: internal/supervisor/cluster.go [hunk 1]
```diff
@@ -73,6 +75,11 @@ func (s *ClusterSupervisor) Do() error {
 		return nil
 	}
 
+	// Sort the clusters by state preference. Relative order is preserved.
+	sort.SliceStable(clusters, func(i, j int) bool {
+		return model.PendingWorkPriority[clusters[i].State] > model.PendingWorkPriority[clusters[j].State]
```
Comment (fmartingr) [inline internal/supervisor/cluster.go:80]: Wouldn't this crash if we try to access an unset State key on the `PendingWorkPriority` map? If we try to `PendingWorkPriority[ClusterStateResizeRequested]`, shouldn't this crash? But I see you added tests and those are passing. I'm confused.

File: model/cluster_states.go [hunk 2]
```diff
@@ -71,6 +71,13 @@ var AllClusterStatesPendingWork = []string{
 	ClusterStateDeletionRequested,
 }
 
+// PendingWorkPriority is a map of states to their priority. Default priority is 0.
+// States with higher priority will be processed first.
+var PendingWorkPriority = map[string]int{
```
Comment (fmartingr) [inline model/cluster_states.go:76]: nit: `ClusterStateWorkPriority` ? 0/5

File: internal/supervisor/cluster_test.go [hunk 3]
```diff
@@ -140,6 +149,61 @@ func TestClusterSupervisorDo(t *testing.T) {
...
+			mockEventProducer,
+			"instanceID",
+			logger,
+		)
+		err := supervisor.Do()
+		require.NoError(t, err)
+
+		clusterListByWorkOrder := mockEventProducer.clusterListByEventOrder
+		require.Equal(t, len(preferredClusterOrder), len(clusterListByWorkOrder))
+		for i, clusterID := range clusterListByWorkOrder {
```
Comment (fmartingr) [inline internal/supervisor/cluster_test.go:202]: I think you could use `Equal` directly to compare that the two slices and it's order are the same.

**Review by fmartingr — COMMENTED** [review]
Looks good! I just have a question about the `map` and what it stores below, to avoid crashes.

File: internal/supervisor/cluster.go (hunk 1 above)
Comment (mirshahriar) [reply internal/supervisor/cluster.go:80]: No. It will return zero value. 

https://go.dev/play/p/koT9sxmFsgP

Comment (fmartingr) [reply internal/supervisor/cluster.go:80]: I may have been thinking on something else then. Good good! Thanks!

File: internal/supervisor/cluster_test.go (hunk 3 above)
Comment (mirshahriar) [reply internal/supervisor/cluster_test.go]: Thanks. That will be nice

**Review by fmartingr — APPROVED** [review]

**Pull request merged**